include/xtensor/xslice.hpp

```diff
@@ -656,8 +629,39 @@ namespace xt
             return xall<std::ptrdiff_t>(static_cast<std::ptrdiff_t>(size));
         }
 
+        A start() const { return m_start; }
+        B stop()  const { return m_stop;  }
+        C step()  const { return m_step;  }
+
     private:
 
+        static auto normalize(const std::ptrdiff_t val,const std::size_t ssize)
+        {
+            std::ptrdiff_t size = static_cast<std::ptrdiff_t>(ssize);
+            val = (val >= 0) ? val : val + size;
+            return (std::max)(std::ptrdiff_t(0), (std::min)(size, val));
+        }
+
+        static auto get_stepped_range(const std::ptrdiff_t start,const std::ptrdiff_t stop,const std::ptrdiff_t step,const std::size_t ssize)
```

Comment:
Same remark here + this prevents compilation.

Reply:
small formatting nitpick: spaces after the commas are the format in the rest of the library.

Reply:
Preparing a new version.